contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,426 @@
+            revert Errors.LicenseRegistry__CallerNotLicensingModule();
+        }
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice initializer for this implementation contract
+    /// @param governance The address of the governance contract
+    function initialize(address governance) public initializer {
+        __GovernableUpgradeable_init(governance);
+        __UUPSUpgradeable_init();
+    }
+
+    /// @dev Sets the DisputeModule address.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param newDisputeModule The address of the DisputeModule
+    function setDisputeModule(address newDisputeModule) external onlyProtocolAdmin {
+        if (newDisputeModule == address(0)) {
+            revert Errors.LicenseRegistry__ZeroDisputeModule();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.disputeModule = IDisputeModule(newDisputeModule);
+    }
+
+    /// @dev Sets the LicensingModule address.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param newLicensingModule The address of the LicensingModule
+    function setLicensingModule(address newLicensingModule) external onlyProtocolAdmin {
+        if (newLicensingModule == address(0)) {
+            revert Errors.LicenseRegistry__ZeroLicensingModule();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.licensingModule = ILicensingModule(newLicensingModule);
+    }
+
+    /// @notice Sets the default license terms that are attached to all IPs by default.
+    /// @param newLicenseTemplate The address of the new default license template.
+    /// @param newLicenseTermsId The ID of the new default license terms.
+    function setDefaultLicenseTerms(address newLicenseTemplate, uint256 newLicenseTermsId) external onlyProtocolAdmin {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.defaultLicenseTemplate = newLicenseTemplate;
+        $.defaultLicenseTermsId = newLicenseTermsId;
+    }
+
+    /// @notice Registers a new license template in the Story Protocol.
+    /// @param licenseTemplate The address of the license template to register.
+    function registerLicenseTemplate(address licenseTemplate) external onlyProtocolAdmin {
+        if (licenseTemplate.supportsInterface(type(ILicenseTemplate).interfaceId)) {
+            revert Errors.LicenseRegistry__NotLicenseTemplate(licenseTemplate);
+        }
+        _getLicenseRegistryStorage().registeredLicenseTemplates[licenseTemplate] = true;
+        emit LicenseTemplateRegistered(licenseTemplate);
+    }
+
+    /// @notice Registers a new royalty policy in the Story Protocol.
+    /// @param royaltyPolicy The address of the royalty policy to register.
+    function registerRoyaltyPolicy(address royaltyPolicy) external onlyProtocolAdmin {
```

Comment:
register royalty policy in license registry?  Is it intended? @Spablob

Reply:
Yes, as the royalty policy is configured in the license anyway. Currently, the royalty policy and currency tokens are registered in the RoyaltyModule, so it is better to move such protocol-wide configuration into the registry. 
However, I have removed the function from the PR to address another comment.
We can address this in a separate PR.